apps/cli/src/commands/SelfAuthCommand.ts

```diff
@@ -0,0 +1,276 @@
+			this.logError("Missing required environment variables:");
+			if (!clientId) console.log("   - LINEAR_CLIENT_ID");
+			if (!clientSecret) console.log("   - LINEAR_CLIENT_SECRET");
+			if (!baseUrl) console.log("   - CYRUS_BASE_URL");
+			console.log("\nSet these in your shell profile (.zshrc):");
+			console.log("  export LINEAR_CLIENT_ID='your-client-id'");
+			console.log("  export LINEAR_CLIENT_SECRET='your-client-secret'");
+			console.log("  export CYRUS_BASE_URL='https://your-tunnel-domain.com'");
+			process.exit(1);
+		}
+
+		// Check config file exists
+		const configPath = resolve(this.app.cyrusHome, "config.json");
+		let config: any;
+		try {
+			config = JSON.parse(readFileSync(configPath, "utf-8"));
+		} catch {
+			this.logError(`Config file not found: ${configPath}`);
+			console.log("Run 'cyrus' first to create initial configuration.");
+			process.exit(1);
+		}
+
+		console.log("Configuration:");
+		console.log(`   Client ID: ${clientId.substring(0, 20)}...`);
+		console.log(`   Base URL: ${baseUrl}`);
+		console.log(`   Config: ${configPath}`);
+		console.log(`   Callback port: ${this.callbackPort}`);
+		console.log();
+
+		try {
+			// Start temporary server to receive OAuth callback
+			const authCode = await this.waitForCallback(clientId);
+
+			// Exchange code for tokens
+			console.log("Exchanging code for tokens...");
+			const tokens = await this.exchangeCodeForTokens(
+				authCode,
+				clientId,
+				clientSecret,
+			);
+			this.logSuccess(
+				`Got access token: ${tokens.accessToken.substring(0, 30)}...`,
+			);
+
+			// Fetch workspace info
+			console.log("Fetching workspace info...");
+			const workspace = await this.fetchWorkspaceInfo(tokens.accessToken);
+			this.logSuccess(`Workspace: ${workspace.name} (${workspace.id})`);
+
+			// Update config.json
+			console.log("Saving tokens to config.json...");
+			this.updateConfig(config, configPath, tokens, workspace);
+
+			const updatedCount = config.repositories.filter(
+				(r: any) => r.linearWorkspaceId === workspace.id,
+			).length;
+			this.logSuccess(`Updated ${updatedCount} repository/repositories`);
+
+			console.log();
+			this.logSuccess(
+				"Authentication complete! Restart cyrus to use the new tokens.",
+			);
+			process.exit(0);
+		} catch (error) {
+			this.logError(`Authentication failed: ${(error as Error).message}`);
+			process.exit(1);
+		} finally {
+			this.cleanup();
+		}
+	}
+
+	private async waitForCallback(clientId: string): Promise<string> {
+		return new Promise((resolve, reject) => {
+			const baseUrl = process.env.CYRUS_BASE_URL;
+			if (!baseUrl) {
+				reject(new Error("CYRUS_BASE_URL environment variable is required"));
+				return;
+			}
+			const redirectUri = `${baseUrl}/callback`;
+			const oauthUrl = `https://linear.app/oauth/authorize?client_id=${clientId}&redirect_uri=${encodeURIComponent(redirectUri)}&response_type=code&scope=read,write,comments:create,app:assignable,app:mentionable&actor=app`;
```

Comment:
our own flow that we've implemented in the hosted version has just:
write,app:assignable,app:mentionable

which seems to be enough, so might want to just leave what's relevant and trim the rest.
can by 'write' write comments